static/app/utils/replays/getFrameDetails.tsx

```diff
@@ -0,0 +1,279 @@
+import {ReactNode} from 'react';
+
+import {Tooltip} from 'sentry/components/tooltip';
+import {IconWarning} from 'sentry/icons';
+import {t, tct} from 'sentry/locale';
+import {BreadcrumbType} from 'sentry/types/breadcrumbs';
+import {TabKey} from 'sentry/utils/replays/hooks/useActiveReplayTab';
+import type {
+  BreadcrumbFrame,
+  ErrorFrame,
+  LargestContentfulPaintFrame,
+  MultiClickFrame,
+  MutationFrame,
+  NavFrame,
+  ReplayFrame,
+  SlowClickFrame,
+} from 'sentry/utils/replays/types';
+import {
+  getFrameOpOrCategory,
+  isDeadClick,
+  isDeadRageClick,
+  isRageClick,
+} from 'sentry/utils/replays/types';
+import type {Color} from 'sentry/utils/theme';
+import stripOrigin from 'sentry/utils/url/stripOrigin';
+
+interface Details {
+  color: Color;
+  description: ReactNode;
+  tabKey: TabKey;
+  title: ReactNode;
+  type: BreadcrumbType; // @deprecated
```

Comment:
Will TS pick this up as a warning? I know it will for jsdoc-style comments, but not sure about inline

Reply:
no i don't think so tbh. i was thinking i could quickly followup and replace this with an `icon` field that we return instead.

Reply:
```suggestion
  type: BreadcrumbType;
```